### PR TITLE
Add a demo notebook

### PR DESCRIPTION
**Pull request by kgryte**
This PR adds a notebook for demonstrating the TOC extension and adds support for launching the demo notebook in Binder.

Having the demo notebook is useful both from the perspective of a potential extension user and from the perspective of the extension developer.

If this PR is approved, I'll modify the Binder URL in the README to point to this repository, rather than my fork, before the final merge.

Potential reviewers: @ian-r-rose

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,6 +40,12 @@ Then launch JupyterLab using
...
 
-This will automatically recompile `@jupyterlab/toc` upon changes,
-and JupyterLab will rebuild itself. You should then be able to refresh the
-page and see your changes.
+This will automatically recompile `@jupyterlab/toc` upon changes, and JupyterLab will rebuild itself. You should then be able to refresh the page and see your changes.
+
+<!-- links -->
+
+[badge-binder]: https://mybinder.org/badge_logo.svg
+[binder]: https://mybinder.org/v2/gh/kgryte/jupyterlab-toc/demo-notebook?urlpath=lab%2Ftree%2Fnotebooks%2Fdemo.ipynb
```
Comment (ian-r-rose) [inline README.md]: This link should be updated to point to this repo.

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (ian-r-rose) [inline .dockerignore]: I don't think there is much reason to include this file in the absence of any docker images.

File: README.md (hunk 1 above)
Comment (ian-r-rose) [reply README.md]: Ah, I see you already mentioned this above :)

File: .dockerignore (hunk 2 above)
Comment (kgryte) [reply .dockerignore]: Removed.

File: README.md (hunk 1 above)
Comment (kgryte) [reply README.md]: Done.

**Comment by kgryte** [comment]
@ian-r-rose Addressed PR feedback. Should be ready for merge.

**Comment by ian-r-rose** [comment]
Looks good!

**Pull request merged**